typed-function.js

```diff
@@ -550,16 +552,16 @@
           // no matching signatures anymore, throw error "wrong type"
           expected = mergeExpectedParams(matchingSignatures, index);
           if (expected.length > 0) {
-            var actualType = findTypeName(args[index]);
+            var actualTypes = Object.keys(findTypeNames(args[index]));
 
             err = new TypeError('Unexpected type of argument in function ' + _name +
                 ' (expected: ' + expected.join(' or ') +
-                ', actual: ' + actualType + ', index: ' + index + ')');
+                ', actual: ' + actualTypes.join(' or ') + ', index: ' + index + ')');
```

Comment:
The join string should be `' and '` (or perhaps `' & '` would be even better) because the actual type of the argument is that it inhabits **all** of those types, i.e., its type is the intersection type of those types.

Reply:
👍  done